### PR TITLE
Replace date picker for entities card

### LINKED ISSUE #4866: Date Picker Display Issues on Chrome (Windows 10)
## Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

## The problem

Google Chrome (desktop version, most recent, Windows 10) is adding webkit spinner arrow buttons to Date Time inputs causing display issues and cutting off part of the text from the dates on a Datetime entity:

image

## Expected behavior

Readable text in Chrome.

## Steps to reproduce

Add a datetime entity row to a standard lovelace card/theme in Google Chrome on Desktop.

## Environment

- Home Assistant release with the issue: 0.105.3
- Last working Home Assistant release (if known): (n/a, checked the git blame, looks like these lines haven't been changed, so it might be a Google Chrome update that broke it).
- UI Type (States or Lovelace): Lovelace
- Browser and browser version: Google Chrome | 80.0.3987.100
- Operating system: Windows 10

## Problem-relevant configuration

```yaml
  - type: entities
    title: Filter Status
    entities:
      - entity: input_datetime.hvac_filter_last_changed
```

## Javascript errors shown in your browser console/inspector

```txt
N/A
```

## Additional information
#3150 Addressed this issue for Firefox.  
Could also be related/resolved if #3920 is progressed.

### PR DESCRIPTION
**Pull request by thomasloven**
## Type of change
Use the vaadin-date-picker we already have for input_datetime in the entities card too.
I made a wrapper to make is ISO compliant and change the formating a bit so it looks more like the paper-input used for time inputs.

I also replaced the date picker in the more-info dialog with the wrapped version.
image

Finally fixes #4866.

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #4866 
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: lazylint [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```
Comment (bramkragten) [inline lazylint:1]: Thanks for this, what can we do with it? 😅

Comment (thomasloven) [reply lazylint:1]: Oh! Oops.

Comment (thomasloven) [reply lazylint]: It does pretty much the same as the core one. Runs the linters on any file that's changed from dev.

**Review by KTibow — APPROVED** [review]

File: src/components/ha-date-input.ts [hunk 2]
```diff
@@ -1,127 +1,69 @@
...
-  @property() public month?: string;
+export class HaDateInput extends VaadinDatePicker {
+  constructor() {
+    super();
 
-  @property() public day?: string;
-
-  @property({ type: Boolean }) public disabled = false;
+    this.i18n.formatDate = this._formatISODate;
+    this.i18n.parseDate = this._parseISODate;
```
Comment (bramkragten) [inline src/components/ha-date-input.ts:10]: We should show the date in the format of the user language, we have functions for that: <https://github.com/home-assistant/frontend/blob/dev/src/common/datetime/format_date.ts#L4>

Comment (bramkragten) [reply src/components/ha-date-input.ts:10]: Something like `formatDateShort(new Date(d.year, d.month, d.day), this.hass.language)`

**Comment by bramkragten** [comment]
Styling in dark mode is wrong:
image

**Comment by bramkragten** [comment]
And can someone remind me why we can't use `<input type="date">`?

**Comment by KTibow** [comment]
Maybe because it might look old or bad on some devices?

**Comment by zsarnett** [comment]
Fing safari man
image

**Comment by bramkragten** [comment]
Fuck safari, let them have text inputs 😛 but then we can't parse that to a data object because safari only accepts ISO 8601 strings....

**Comment by zsarnett** [comment]
I don't understand why they have to make literally everything so Difficult

**Comment by bramkragten** [comment]
It's weird that we don't have a `localDateStringToDate` function...

**Comment by thomasloven** [comment]
image
Theming has been fixed.

Parsing date strings according to locale seems way harder than you'd think at first glance. Especially since we allow several formats for this; DD, MM-DD, YY-MM-DD and YYYY-MM-DD.
Then there's the order, separators, pre- and postfixes to keep track of. And that's just for those that use the Julian calendar...

**Comment by thomasloven** [comment]
There's still an annoying 5 px difference in alignment between date and time numbers, but I didn't want to ugly-hack it away, and anything better would require getting more vaadin stuff imported - which I believe we try to avoid? The date picker is used here because we already had it anyway.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**